rnd/autogpt_server/autogpt_server/server/server.py

```diff
@@ -200,6 +229,35 @@ async def websocket_endpoint(websocket: WebSocket):  # type: ignore
 
         uvicorn.run(app, host="0.0.0.0", port=8000)
 
+    def set_test_dependency_overrides(self, overrides: dict):
+        self._test_dependency_overrides = overrides
+
+    def _apply_overrides_to_methods(self):
+        for attr_name in dir(self):
+            attr = getattr(self, attr_name)
+            if callable(attr) and hasattr(attr, "__annotations__"):
+                setattr(self, attr_name, self._override_method(attr))
+
+    # TODO: fix this with some proper refactoring of the server
+    def _override_method(self, method):
```

Comment:
This is not ideal, but because of how we're setting up app and running AgentServer, I could only get dependency overrides to work this way